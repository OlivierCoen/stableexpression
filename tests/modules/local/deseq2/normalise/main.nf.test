nextflow_process {

    name "Test Process DESEQ2_NORMALISE"
    script "modules/local/deseq2/normalise/main.nf"
    process "DESEQ2_NORMALISE"
    tag "deseq2"
    tag "normalise"
    tag "module"

    test("Very small dataset - no zeros") {

        when {

            process {
                """
                meta = [accession: "accession", design: file('$projectDir/tests/input/normalise/design.csv')]
                input[0] = [meta, file('$projectDir/tests/input/normalise/all_counts.csv')]
                input[1] = false
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.csv).match()
        }

    }

    test("Very small dataset - with zeros") {

        when {

            process {
                """
                meta = [accession: "accession", design: file('$projectDir/tests/input/normalise/design.csv')]
                input[0] = [meta, file('$projectDir/tests/input/normalise/all_counts.csv')]
                input[1] = true
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.csv).match()
        }

    }

    test("Rows with many zeros") {

        when {

            process {
                """
                meta = [accession: "accession", design: file('$projectDir/tests/input/normalise/design_rows_many_zeros.csv')]
                input[0] = [meta, file('$projectDir/tests/input/normalise/counts_rows_many_zeros.csv')]
                input[1] = true
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.csv).match()
        }

    }

}
