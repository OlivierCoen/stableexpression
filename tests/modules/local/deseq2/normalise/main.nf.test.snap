{
    "Very small dataset": {
        "content": [
            [
                [
                    {
                        "accession": "accession",
                        "design": "design.csv:md5,a83dd6a15463b51d94f0a42c196d7933"
                    },
                    "counts.log_cpm.csv:md5,4091ece4324283f6edb3b91fafa324fa"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-03T12:53:09.580478043"
    },
    "One group": {
        "content": [
            [
                [
                    {
                        "accession": "accession",
                        "design": "design.csv:md5,28cf54802df5df4a9dc406003623c6a7"
                    },
                    "counts.log_cpm.csv:md5,5c662f9c6a7422119ba94b61b4bf695d"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-03T12:53:48.817205569"
    },
    "Very small dataset - no zeros": {
        "content": [
            [
                [
                    {
                        "accession": "accession",
                        "design": "design.csv:md5,a83dd6a15463b51d94f0a42c196d7933"
                    },
                    "all_counts.log_cpm.csv:md5,4091ece4324283f6edb3b91fafa324fa"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T12:30:02.920269"
    },
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "accession": "accession",
                        "design": "design.csv:md5,a83dd6a15463b51d94f0a42c196d7933"
                    },
                    "all_counts.log_cpm.csv:md5,4091ece4324283f6edb3b91fafa324fa"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-28T22:30:31.238971692"
    },
    "Rows with many zeros": {
        "content": [
            [
                [
                    {
                        "accession": "accession",
                        "design": "design.csv:md5,3cdf98e2e202b4af2687eaefd9bdd8e9"
                    },
                    "counts.log_cpm.csv:md5,a5640eeb84b90577e593ef0ca906f937"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-03T12:53:29.441742091"
    },
    "Very small dataset - with zeros": {
        "content": [
            [
                [
                    {
                        "accession": "accession",
                        "design": "design.csv:md5,a83dd6a15463b51d94f0a42c196d7933"
                    },
                    "all_counts.log_cpm.csv:md5,4091ece4324283f6edb3b91fafa324fa"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T12:20:21.710139"
    }
}